snuba/datasets/entities/sessions.py

```diff
@@ -99,6 +100,18 @@ def __init__(self) -> None:
             abstract_column_set=read_schema.get_columns(),
             join_relationships={},
             writable_storage=writable_storage,
+            required_conditions={
+                "org_id": [ConditionFunctions.EQ],
```

Comment:
Would we want to check the type of the value we're comparing to? Like to ensure org_id is an int in this case, and not some other kind of expression

Reply:
I can. I left that off for two reasons: 1) we haven't started doing that type of checking yet 2) The query will fail if you pass a bad argument, and these are here to prevent valid queries that run over too much data.

Reply:
The query might still be valid as far as ClickHouse is concerned though, and cause ClickHouse to read a lot of data.
e.g. `org_id = key_id`

Reply:
I'd agree with @lynnagara  on this. For conditions like project and organization, being overly strict is a good idea so I would also validate the right hand side of the condition to be of the right type. I am ok in doing it in a followup if it adds too much complexity in this one.

Reply:
The matchers now do this.